processing/data_collection/gazette/spiders/ma_sao_luis.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+import dateparser
+from datetime import datetime
+from scrapy_splash import SplashRequest
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+from scrapy.shell import inspect_response
+
+
+class MaSaoLuisSpider(BaseGazetteSpider):
+    MUNICIPALITY_ID = '2111300'
+    name = 'ma_sao_luis'
+    allowed_domains = ['www.semad.saoluis.ma.gov.br']
+    start_urls = ['http://www.semad.saoluis.ma.gov.br:8090/easysearch/']
+
+    lua_script = """
+function wait_for_element(splash, css, value, maxwait)
+  -- Wait until a selector matches an element
+  -- and it contains a specific value
+  -- in the page. Return an error if waited more
+  -- than maxwait seconds.
+  if maxwait == nil then
+      maxwait = 10
+  end
+  return splash:wait_for_resume(string.format([[
+    function main(splash) {
+      var selector = '%s';
+      var value = %s;
+      var maxwait = %s;
+      var end = Date.now() + maxwait*1000;
+      function check() {
+        if(document.querySelector(selector) && document.querySelector(selector).value === value) {
+          splash.resume('Element found');
+        } else if(Date.now() >= end) {
+          var err = 'Timeout waiting for element';
+          splash.error(err + " " + selector);
+        } else {
+          setTimeout(check, 200);
+        }
+      }
+      check();
+    }
+  ]], css, value, maxwait))
+end
+
+function main(splash, args)
+    getElementByXPath = splash:jsfunc([[
+    function (path) {
+        return document.evaluate(path, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue;
+    }
+    ]])
+    splash:go(args.url)
+    wait_for_element(splash, 'a.campoResultadoDownload', '1', 40)
+    page_input_box = getElementByXPath("//div[@class='GPFMNGWNP']/input[@class='gwt-TextBox GPFMNGWOJ GPFMNGWIK']")
+    page_input_box:send_keys("<Delete>")
+    page_input_box:send_text(args.page_number)
+    page_input_box:send_keys("<Return>")
+    wait_for_element(splash, 'a.campoResultadoDownload', args.page_number, 40)
+    return {html = splash.html(), cookies = splash:get_cookies(), png = splash.png()}
+end
+"""
+
+    def parse(self, response):
```

Comment:
Why you didn't use the `start_requests` to make these requests? It makes more sense as you are making initial requests and not parsing (as `parse` say) a page. This is the recommended pattern for Scrapy.